scijava/scijava-ops/src/main/java/org/scijava/ops/hints/impl/DefaultHints.java

```diff
@@ -15,16 +17,23 @@
 public class DefaultHints extends AbstractHints {
 
 	public DefaultHints() {
-		super(new String[] {Simplification.ALLOWED, Adaptation.ALLOWED});
+		super(new String[] {Simplification.ALLOWED, Adaptation.ALLOWED, DependencyMatching.NOT_IN_PROGRESS});

```

Comment:
Hmm, I'm not sure that I like that move. We need to `export` the `Simplification` pseudo-enum, but I'd prefer not to export the package `org.scijava.ops.simplify` (later to become `org.scijava.ops.engine.simplify`). I *suppose* we could rename `org.scijava.ops.engine.simplify` ➡️ `org.scijava.ops.engine.simplify.impl`, then make a new package `org.scijava.ops.engine.simplify` with _just_ `SimplificationHints`? That doesn't seem all too bad to me...

(Of course, we'd do the same thing for Adaptation, and later Reduction)

Reply:
I thought the `scijava-ops-engine` module was implementation-only, and exports nothing? No? If so, I'd like to keep it that way. Or if not, I'd like to understand better what it's exporting (presumably for compile-time dependencies downstream).

I think the hints pseudo-enums need to go into the `scijava-ops-api` module. Which package is something to discuss—obviously an exported one. :smile:

Reply:
> Or if not, I'd like to understand better what it's exporting (presumably for compile-time dependencies downstream).

There are three packages exported:
* `org.scijava.ops.engine`: Contains `OpService` / `OpHistoryService`, which will later go away, and some other API classes that could presumably go into `scijava-ops-api`. And then `BaseOpHints`, which contains the pseudo-enum values
* `org.scijava.ops.engine.conversionLoss`: Contains the `LossReporter` interface, which is an implementation detail of the Simplification framework
* `org.scijava.ops.engine.util`: Contains utility methods useful to Op implementations. I'd like to delete this package at some point.

> I think the hints pseudo-enums need to go into the `scijava-ops-api` module.

I disagree. They all pertain to matching, which is restricted to the `scijava-ops-engine` module. `scijava-ops-api` should not concern itself with these implementation details.

> (does `scijava-ops-api` even have any internal packages?).

Nope :grin:

Reply:
What about merging `OpService` and `OpHistoryService`, and moving them to `scijava-ops-services`? This would be a step forward toward eliminating the scijava-common dependency from `scijava-ops-engine`. (It is a necessary but not sufficient step.) Not necessarily now, but let's discuss further after the PR chain is complete.

Reply:
@ctrueden do 1f0d56a3 and fb5e5037, combined with scijava/scijava#71 and scijava/scijava#72 cover everything we discussed here?

Reply:
Yep, looks good! The only other related thing we discussed, which should perhaps be another issue, is the idea that `scijava-ops-api` contains more than one layer of abstractions, which could conceivably be split into two modules later if we decide it would be nicer: 1) the base layer of interfaces for people who want to just use the `OpEnvironment`, and the more specialized interfaces pertaining to features the majority of users won't care about explicitly: adaptation, simplification, loss reporting, etc. My preference, for now, would be to delineate these more specialized interfaces into one or more subpackages of `.api`, so that it is clear they are less central than then things in `.api` directly. What do you think? For the moment we can just file an issue.